### PR TITLE
feat(alert, button, chip, inline-editable, input-message, link, scrim): move theme attribute styles to global.scss

### PR DESCRIPTION
**Pull request by caripizza**
**Related Issues:** #1737, #1537, #1500

## Summary
This pr moves remaining components' styles which hooked directly into the `theme` attribute over to the global stylesheet instead. It adds e2e tests for their theme styles, and also updates calcite-chip's styles to match what I see in Figma (and removes that Sass loop for the colors).

### CODE REVIEW

File: src/assets/styles/global.scss [hunk 1]
```diff
@@ -20,6 +20,17 @@
...
+  --calcite-button-transparent-hover: rgba(0, 0, 0, 0.05);
+  --calcite-button-transparent-press: rgba(0, 0, 0, 0.08);
+  --calcite-chip-solid-bg-light: #{rgba($ui-text-inverse-light, 0.1)};
+  --calcite-chip-solid-bg-dark: #{rgba($ui-text-inverse-dark, 0.1)};
+  --calcite-chip-font-color: #{$ui-text-1-light};
+  --calcite-chip-transparent-hover: rgba(0, 0, 0, 0.05);
+  --calcite-chip-transparent-press: rgba(0, 0, 0, 0.08);
+  --calcite-input-message-floating-background: #{rgba($ui-foreground-1-light, 0.96)};
+  --calcite-link-blue-underline: #{rgba($h-bb-070, 0.4)};
+  --calcite-scrim-background: #{rgba($blk-000, 0.75)};
```
Comment (caripizza) [inline src/assets/styles/global.scss:33]: chip and button use the same transparent colors here, should we combine?

Comment (bstifle) [reply src/assets/styles/global.scss:33]: chips should not have any transparency

**Comment by bstifle** [comment]
> @bstifle can you let me know how chip's new styles look here based on Figma?

There shouldn't be any chips with transparency

**Comment by caripizza** [comment]
calcite-modal renders a calcite-scrim with [`theme="dark"`](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-modal/calcite-modal.tsx#L125), so Screener picked up this style change because the `--calcite-scrim-background` CSS property is now nested inside calcite-modal that has the default non-dark theme (ie., light).

Is this intended? Or should calcite-scrim just have the dark theme background style in general?

**Comment by paulcpederson** [comment]
scrim for modal should always be dark, a white scrim in dark mode doesn't make sense, IMO

**Comment by driskull** [comment]
Nice work @caripizza! Looks good to me.

> Are you saying the close button shouldn't have transparency when you hover or focus it?

It's safer to not use transparency here since it could affect the a11y readability when hovered or focused. Could we use a text-color/background change instead?

File: src/assets/styles/global.scss (hunk 1 above)
Comment (bstifle) [reply src/assets/styles/global.scss:33]: yep my bad

**Comment by bstifle** [comment]
> Nice work @caripizza! Looks good to me.
> 
> > Are you saying the close button shouldn't have transparency when you hover or focus it?
> 
> It's safer to not use transparency here since it could affect the a11y readability when hovered or focused. Could we use a text-color/background change instead?

Thats not accurate. Hover states don't actually need to pass A11y, even so it is being applied behind the X, and graphical elements need only be 3:1

**Comment by bstifle** [comment]
> Could we use a text-color/background change instead?


We could use a hex value. but we dont have any variables for lighter/darker greens/yellows anymore

**Comment by driskull** [comment]
> Thats not accurate. Hover states don't actually need to pass A11y, even so it is being applied behind the X, and graphical elements need only be 3:1

My bad!

File: src/assets/styles/global.scss [hunk 2]
```diff
@@ -20,6 +20,12 @@
   --calcite-ui-opacity-disabled: 0.5;
 
   --calcite-panel-width-multiplier: 1;
+  --calcite-alert-dismiss-progress-background: #{rgba($blk-000, 0.8)};
```
Comment (paulcpederson) [inline src/assets/styles/global.scss:23]: @bstifle I wonder if in the future we should just add a sequence of transparent blacks to the palette? Seems like most of these are just various amounts of semi-transparent black

**Review by paulcpederson — APPROVED** [review]

**Pull request merged**